paddle/phi/infermeta/spmd_rules/c_embedding.h

```diff
@@ -0,0 +1,43 @@
+/* Copyright (c) 2024 PaddlePaddle Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License. */
+
+#pragma once
+
+#include "paddle/phi/core/distributed/auto_parallel/dist_meta_tensor.h"
+#include "paddle/phi/core/distributed/type_defs.h"
+
+namespace phi {
+namespace distributed {
+
+SpmdInfo CEmbeddingInferSpmd(const DistMetaTensor& weight,
+                             const DistMetaTensor& x,
+                             int start_index,
+                             int vocab_size);
+
+SpmdInfo CEmbeddingInferSpmdBase(const DistMetaTensor& weight,
+                                 const DistMetaTensor& x,
+                                 int64_t start_index,
+                                 int64_t vocab_size);
+
+SpmdInfo CEmbeddingGradInferSpmd(const DistMetaTensor& weight,
+                                 const DistMetaTensor& x,
+                                 const DistMetaTensor& out_grad,
+                                 int start_index);
+
+SpmdInfo CEmbeddingGradInferSpmdBase(const DistMetaTensor& weight,
+                                     const DistMetaTensor& x,
+                                     const DistMetaTensor& out_grad,
+                                     int64_t start_index);
```

Comment:
Remove redundant functions `CEmbeddingGradInferSpmdBase` and `CEmbeddingInferSpmdBase`